e2e/cypress/integration/at_mentions/at_mentions_user_spec.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Mention user', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M19761 autocomplete should match on spaces', () => {
+        cy.get('#post_textbox').type('@Samuel Tuc');
+        cy.get('.mention__fullname').should('have.text', 'Samuel Tucker');
```

Comment:
This fails in case the test user has nickname set.  As mentioned by @M-ZubairAhmed, you may use `cy.findByText`. I can see tests something like, (feel free to add/modify test cases as you see fit):
```javascript
function verifySuggestionList({input, expected, withoutSuggestion}) {
    var re = new RegExp(expected, 'g');

    cy.get('#post_textbox').clear().type(input);

    if (withoutSuggestion) {
        cy.get('#suggestionList').should('not.exist');
    } else {
        cy.get('#suggestionList').should('be.visible').within(() => {
            cy.findByText(re).should('be.visible');
        });
    }
}

[
    {input: '@samuel.tucker', expected: 'Samuel Tucker'},
    {input: '@samuel', expected: 'Samuel Tucker'},
    {input: '@tucker', expected: 'Samuel Tucker'},
    {input: '@Samuel', expected: 'Samuel Tucker'},
    {input: '@Tucker', expected: 'Samuel Tucker'},
    {input: '@Samuel Tuc', expected: 'Samuel Tucker'},
    {input: '@Samuel Tucker', expected: 'Samuel Tucker'},
    {input: '@Samuel Tucker ', expected: 'Samuel Tucker', withoutSuggestion: true},
].forEach((testCase) => {
    verifySuggestionList(testCase);
});
```

Reply:
very elegant way of doing it. i like it